game-core/src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java

```diff
@@ -1652,14 +1652,13 @@ private boolean getCanAllUnitsWithRequiresUnitsBePlacedCorrectly(
     if (to == null) {
       return new ArrayList<>();
     }
-    final Collection<Unit> unitsInTo = to.getUnits();
     final Collection<Unit> unitsPlacedAlready = getAlreadyProduced(to);
     if (Matches.territoryIsWater().test(to)) {
       for (final Territory current : getAllProducers(to, player, null, true)) {
         unitsPlacedAlready.addAll(getAlreadyProduced(current));
       }
     }
-    final Collection<Unit> unitsAtStartOfTurnInTo = new ArrayList<>(unitsInTo);
+    final Collection<Unit> unitsAtStartOfTurnInTo = new ArrayList<>(to.getUnits());
```

Comment:
The existing code seems to have been kinda not optimal, returning a copy of a list to then create a copy of a list..

More fundamentally, if this is a heavily invoked code path, wouldn't the better datastructure here be a `Set` instead of `List` for faster item removal?